packages/contractkit/src/wrappers/BaseWrapper.ts

```diff
@@ -50,19 +50,22 @@ export const valueToInt = (input: BigNumber.Value) =>
 export const valueToFrac = (numerator: BigNumber.Value, denominator: BigNumber.Value) =>
   valueToBigNumber(numerator).div(valueToBigNumber(denominator))
 
-export const stringToBuffer = hexToBuffer
-
-export const bufferToString = (buf: Buffer) => ensureLeading0x(buf.toString('hex'))
-
-type SolBytes = string | number[]
-const toBytes = (input: any): SolBytes => input
-const fromBytes = (input: SolBytes): any => input as any
-
-export const stringToBytes = (input: string) => toBytes(ensureLeading0x(input))
-
-export const bufferToBytes = (input: Buffer) => stringToBytes(bufferToString(input))
-
-export const bytesToString = (input: SolBytes): string => fromBytes(input)
+// Type of bytes in solidity gets repesented as a string of number array by typechain and web3
+// Hopefull this will improve in the future, at which point we can make improvements here
+type SolidityBytes = string | number[]
+export const stringToSolidityBytes = (input: string) => ensureLeading0x(input) as SolidityBytes
+export const bufferToSolidityBytes = (input: Buffer) => stringToSolidityBytes(bufferToHex(input))
+export const solidityBytesToString = (input: SolidityBytes): string => {
+  if (input === null || input === undefined || typeof input === 'string') {
+    return input
+  } else if (Array.isArray(input)) {
+    let hex = ''
+    input.map((num) => (hex += num))
```

Comment:
This is not a detail, haha
What characteristic has each number? it's always a one-digit?
Otherwise, if every number it's a two-digit, you should handle the zero padding
```
hex +=  num.toString().padStart(2, '0')
```
Otherwise [88, 1, 50] will be '88150', and probably should be '880150'

Reply:
oh, and even more, this won't handle the `10` as `A`
Probably (we have to check this) the real solution is:
```
hex +=  num.toString(16).padStart(2, '0')
```
And [23, 1, 10, 15] will be '17010A0F'

Reply:
Good questions here. Unfortunately, I'm not sure what the actual format of the data is when web3 gives a `number[]`. It doesn't seem to be documented and I've just pieced together some information based on these discussions:
https://github.com/ethereum-ts/TypeChain/issues/98
https://github.com/ethereum-ts/TypeChain/pull/153
https://github.com/ethereum-ts/TypeChain/issues/123

Reply:
And in my testing, I've only seen it return string. Maybe we should log and throw an error here until we find a situation that actually hits this case? Then we can inspect the number[] and see how this should be implemented. WDYT?

Reply:
Ok, reading some other questions, I think that this case is happening when is used an array of `bytes` in solidity.
And it would probably be a `number` for every `byte`, so I'm 98% sure (haha) that this:
`hex +=  num.toString(16).padStart(2, '0')` will work, otherwise it's impossible to have a letter in the hex, and that would be really weird.

This was probably not working before. My only problem with adding something that will fail if the array was used, is that we don't know if the user will know how to handle that.

In my point of view, in these cases, I prefer to follow what could be a convention in transforming array of numbers into hex, and if we are doing something wrong, this would be an issue from some functionality using it.

Reply:
I found something. The lib `eth-lib` that we use, has this function:
```
const fromArray = arr => {
  let hex = "0x";
  for (let i = 0, l = arr.length; i < l; ++i) {
    let b = arr[i];
    hex += (b < 16 ? "0" : "") + b.toString(16);
  }
  return hex;
}
```
that basically transform an array of integers, the same way we are saying.
If I have to strictly think of an array of numbers how it turns to a hex, this is the only way possible.

Reply:
Thanks for investigating! I'll go ahead and use a hex conversion approach similar to your suggestion and that snippet :)